### PR TITLE
Validate label selectors in ValidateCreate

### PR DESCRIPTION
**Pull request by ptnapoleon**
## What does this PR do?

- [ ] Adds new functionality
- [ ] Alters existing functionality
- [ ] Fixes a bug
- [ ] Improves documentation or testing

Please briefly describe your changes as well as the motivation behind them:
- A user ran into a runtime problem when they used invalid label selector characters. Given that these can be validated at create time, we should do so, to save the user time and confusion

## Code Quality Checklist

- [ ] The documentation is up to date.
- [ ] My code is sufficiently commented and passes continuous integration checks.
- [ ] I have signed my commit (see [Contributing Docs](../CONTRIBUTING.md)).

## Testing

- [ ] I leveraged continuous integration testing
    - [ ] by depending on existing `unit` tests or `end-to-end` tests.
    - [ ] by adding new `unit` tests or `end-to-end` tests.
- [ ] I manually tested the following steps:
    - `x`
    - [ ] locally.
    - [ ] as a canary deployment to a cluster.

### CODE REVIEW

File: api/v1beta1/disruption_types.go [hunk 1]
```diff
@@ -320,13 +321,60 @@ func (s DisruptionSpec) Validate() (retErr error) {
 	return multierror.Prefix(retErr, "Spec:")
 }
 
+func AdvancedSelectorsToRequirements(advancedSelectors []metav1.LabelSelectorRequirement) ([]labels.Requirement, error) {
```
Comment (aymericDD) [inline api/v1beta1/disruption_types.go:324]: If this function is public, we should add dedicated tests and comment it.

File: api/v1beta1/disruption_types.go [hunk 2]
```diff
@@ -320,13 +321,60 @@ func (s DisruptionSpec) Validate() (retErr error) {
...
+		case metav1.LabelSelectorOpIn:
+			op = selection.In
+		case metav1.LabelSelectorOpNotIn:
+			op = selection.NotIn
+		case metav1.LabelSelectorOpExists:
+			op = selection.Exists
+		case metav1.LabelSelectorOpDoesNotExist:
+			op = selection.DoesNotExist
+		default:
+			return nil, fmt.Errorf("error parsing advanced selector operator %s: must be either In, NotIn, Exists or DoesNotExist", req.Operator)
```
Comment (aymericDD) [inline api/v1beta1/disruption_types.go:341]: Do we need to test the invalid operator case?

**Comment by ptnapoleon** [comment]
sorry about the force push, I accidentally merged main with the github UI w/o pushing my local changes yet (I was stuck on the gpg prompt and didnt notice)

**Review by aymericDD — APPROVED** [review]
Nice 👌

**Pull request merged**